clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -222,7 +222,14 @@
     private static final String EXCLUDE_INTERNAL_TOPICS_DOC = "Whether records from internal topics (such as offsets) should be exposed to the consumer. "
                                                             + "If set to <code>true</code> the only way to receive records from an internal topic is subscribing to it.";
     public static final boolean DEFAULT_EXCLUDE_INTERNAL_TOPICS = true;
-    
+
+    /**
+     * <code>leave.group.on.close</code>
+     * Whether or not the consumer should leave the group on close. If set to <code>false</code> then a rebalance
+     * won't occur until <code>session.timeout.ms</code> expires.
+     */
+    static final String LEAVE_GROUP_ON_CLOSE_CONFIG = "leave.group.on.close";
```

Comment:
Yeah, sorry @ijuma. The original idea was to make it public, but based on discussions with
@hachikuji, @guozhangwang, @ewencp, there were some concerns about how generally useful it is for other libraries. 
In streams we also have our custom partition assignor that tries to favour task stickyness. So for us this solution works well. However without that then it is likely to be less useful.
@hachikuji may have more to add